### PR TITLE
Optional relationals

### PR DESCRIPTION
**Pull request by mtydykov**
Modify Arithmetic classes so that operations with optionals result in an optional type and add associated tests.

### CODE REVIEW

**Comment by llllllllll** [comment]
looks like we need to import pytest

**Comment by mtydykov** [comment]
Yup, sorry. I removed that mistakenly. It's back now.

**Comment by llllllllll** [comment]
Can you add a new entry to the whatsnew?

**Comment by mtydykov** [comment]
Sure - which file to I add it to? For the latest version, there's BASE, LOCAL, and REMOTE.

**Comment by llllllllll** [comment]
docs/source/whatsnew/0.9.0.txt

**Comment by mtydykov** [comment]
Done.

File: blaze/expr/tests/test_arithmetic.py [hunk 1]
```diff
@@ -27,12 +34,26 @@ def test_unary_ops_are_elemwise():
...
+
+@pytest.mark.parametrize('relation_type', Relational.__subclasses__())
+def test_relations_maintain_shape(relation_type):
+    assert relation_type(x, y).shape == x.shape
 
 
-def test_relations_are_boolean():
-    assert Gt(x, y).schema == dshape('bool')
+@pytest.mark.parametrize('relation_type', Relational.__subclasses__())
+@pytest.mark.parametrize('lhs,rhs', product((x, y, d, e, f), repeat=2))
```
Comment (cpcloud) [inline blaze/expr/tests/test_arithmetic.py]: neat! i didn't realize you could chain these

**Comment by cpcloud** [comment]
i'm going to resolve the merge conflicts here and merge on passing

**Pull request merged**